### PR TITLE
Simplified attribute docstrings

### PR DESCRIPTION
**Pull request by mapazarr**
While reviewing #299 by @mapazarr I noticed that the documentation for attributes is very verbose and basically repeats the same info twice:

screen shot 2015-07-15 at 11 13 00

I think I'd prefer we don't use `Returns` sections for attributes, but instead just have one line that also mentions the return type like I've done it here:
https://gammapy.readthedocs.org/en/latest/api/gammapy.data.EventList.html
If more explanations are needed / wanted, that's OK, extra lines can be added.
My point is only that `Returns` doesn't make much sense because attributes always return exactly one thing, and that's already described in the first line.

@mapazarr @adonath @kingj90 – What do you think?

If we agree on this, the TODOs here are:
- [x] Add an entry explaining this convention here: https://gammapy.readthedocs.org/en/latest/development/index.html
- [x] Go through the existing attributes and clean them up. E.g. [SpectralCube](https://gammapy.readthedocs.org/en/latest/api/gammapy.data.SpectralCube.html) ... there don't seem to be too many: https://github.com/gammapy/gammapy/search?utf8=%E2%9C%93&q=property

### CODE REVIEW

**Comment by mapazarr** [comment]
Ok, I reformatted all attribute docstrings following the example on the [EventList](https://gammapy.readthedocs.org/en/latest/api/gammapy.data.EventList.html) class.
Can we merge it?

**Comment by cdeil** [comment]
I looked for < 1 min and it looks OK.
@adonath – Can you do a little more careful final review for this?

**Comment by cdeil** [comment]
Also, @mapazarr: thanks for fixing / cleaning up docstring and minor code issues as you encounter them. Gammapy needs a ton of cleanup / polishing because we have a lot of legacy code that never went through code review.

**Comment by cdeil** [comment]
I've put myself as assignee ... I'll review this later this week.

**Comment by cdeil** [comment]
@mapazarr Please rebase this branch against master.

File: gammapy/background/kernel.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def mask_image_hdu(self):
 
     @property
     def background_image_hdu(self):
-        """Returns resulting background estimate as `~astropy.io.fits.ImageHDU`."""
+        """Resulting background estimate (`~astropy.io.fits.ImageHDU`)"""
```
Comment (cdeil) [inline gammapy/background/kernel.py]: Remove "resulting" (also below) ... it doesn't give useful info.

File: gammapy/background/models.py [hunk 2]
```diff
@@ -561,12 +539,12 @@ def find_energy_bin(self, energy):
 
         Parameters
         ----------
-        energy : `~numpy.array`
+        energy : `~numpy.ndarray`
```
Comment (cdeil) [inline gammapy/background/models.py]: This parameter should be quantity , not ndarray, no?

File: gammapy/background/models.py [hunk 3]
```diff
@@ -532,12 +510,12 @@ def find_det_bin(self, det):
 
         Parameters
         ----------
-        det : `~numpy.array`
+        det : `~numpy.ndarray`
```
Comment (cdeil) [inline gammapy/background/models.py]: Give Angle as the type:
```python
det : `~astropy.coordinates.Angle`
   Array of det (x, y) pairs to search for.
```

File: gammapy/background/kernel.py (hunk 1 above)
Comment (mapazarr) [inline gammapy/background/kernel.py]: Done.

File: gammapy/background/models.py (hunk 2 above)
Comment (mapazarr) [inline gammapy/background/models.py]: Done.

File: gammapy/background/models.py (hunk 3 above)
Comment (mapazarr) [inline gammapy/background/models.py]: Done.

**Comment by mapazarr** [comment]
@cdeil OK I rebased again and implemented the comments. You can merge it.

**Comment by cdeil** [comment]
Thanks!
Merging now.

I'll change two things in `master` later today:
- Add one or two examples to `docs/development/index.rst` and link to a few existing examples. The goal of that page is a reminder and help to devs, so code examples they can quickly copy & paste from and use as templates is better than just descriptions.
- Change the wording for tuples a bit ... I don't like the "2x" ... will have to think of something better, not sure yet.

**Comment by cdeil** [comment]
@mapazarr – I've expanded the documentation guidelines section a bit and added examples: 2f5edf1

It took me a while to understand the issue ... actually the short format isn't specific to properties, it should be used for all functions / methods that return a single value.

Concerning the documentation (or more generally the use of) of attributes and properties, I've left a comment and TODO, because I haven't come up with a good guideline yet.

**Pull request merged**